src/routes/UserSettings/EditProfileTab/EditProfileTab.js

```diff
@@ -159,76 +237,30 @@ export default class EditProfileTab extends Component {
       <label styleName='social-label'>Social Accounts</label>
       <SocialControl
         label='Facebook'
-        onLink={() => loginWithService('facebook')}
-        onChange={this.updateSettingDirectly('facebookUrl', false)}
-        unlinkAccount={unlinkAccount}
         provider='facebook'
-        value={facebookUrl} />
+        value={facebookUrl}
+        onLink={() => loginWithService('facebook')}
+        updateSocialSetting={this.updateSocialSetting}
+        handleUnlinkAccount={() => unlinkAccount('facebook')}
```

Comment:
I chose to pass `unlinkAccount` with the appropriate provider value filled in so that the child component is "dumber" and only needs to invoke `handleUnlinkAccount()`. 

I'm not super wed to this though since we still need to pass in provider for the switch/case. I've just been told its good to keep child components dumb!

Reply:
sounds good to me!